eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala

```diff
@@ -162,7 +164,13 @@ class PeerConnection(keyPair: KeyPair, conf: PeerConnection.Conf, switchboard: A
     heartbeat {
       case Event(msg: LightningMessage, d: ConnectedData) if sender() != d.transport => // if the message doesn't originate from the transport, it is an outgoing message
         d.transport forward msg
-        stay()
+        msg match {
+          // If we send any channel management message to this peer, we enable the gossip.
+          case _ : ChannelMessage if !d.enableGossip =>
+            context.system.eventStream.subscribe(self, classOf[Rebroadcast])
+            stay() using d.copy(enableGossip = true)
+          case _ => stay()
+        }
```

Comment:
This is clever, I initially thought we would need to do it in the `Peer`, which isn't great because that would require sending messages from `PeerConnection` (potentially on front machines) to `Peer` (on backend machines). I'm glad you thought of this trick, it's better to do this here.

However matching on any `ChannelMessage` is too broad and easily gameable by our peer: they could send us a dummy `open_channel` and not follow through, and we'd enable the gossip. I think we should instead match on `channel_reestablish` and `funding_locked` to enable the gossip:

- if we send an outgoing `channel_reestablish`, it means we previously had a channel
- if we send an outgoing `funding_locked`, it means that we've seen a funding tx broadcast, so there really is a new channel being created